packages/admin-next/dashboard/src/routes/products/product-edit-variant/loader.ts

```diff
@@ -0,0 +1,40 @@
+import { adminProductKeys, adminStoreKeys } from "medusa-react"
+import { LoaderFunctionArgs } from "react-router-dom"
+
+import { medusa, queryClient } from "../../../lib/medusa"
+
+const queryKey = (id: string) => {
+  return [adminProductKeys.detail(id), adminStoreKeys.details()]
+}
+
+const queryFn = async (id: string) => {
+  const productRes = await medusa.admin.products.retrieve(id)
```

Comment:
suggestion: do we need to fetch the product here...seems that fetching only the variant we are editing is sufficient

Reply:
Yes, we need it for the options editing (not implemented here yet). The variant does not hold the name, or the other options in product.